### PR TITLE
[TRTLLM-11320][refactor] Refactor VisualGenArgs API and registry

### PR DESCRIPTION
**Pull request by zhenhuaw-me**
## Description

Refactor the VisualGen public API surface (TRTLLM-11320 and follow-up cleanups) so the schema, sub-config naming, pipeline registry, and CLI flags are validated in one place.

- Move `VisualGenArgs` and its sub-configs to `tensorrt_llm.visual_gen.args`. Rename master switches to drop class-name prefixes (`dit_cfg_size` → `cfg_size`, …) and add `_config` suffixes to sub-config attributes (`attention` → `attention_config`, …).
- Clean up the schema: replace flat `skip_warmup` / `skip_components` with `CompilationConfig.skip_warmup` and component-level controls; resolve quant lazily in `PipelineLoader`; reshape `SageAttentionConfig` into a recipe-based `QuantAttentionConfig` validated centrally.
- Add a per-family pipeline registry, exposed via `VisualGen.supported_models()` / `VisualGen.pipeline_config(model)` and validated at load time. LTX-2 paths now live inside `pipeline_config`.
- Add `--visual_gen_args` on `trtllm-serve` / `trtllm-bench`, with `--extra_visual_gen_options` retained as a back-compat alias.

Example YAML configs and example scripts are updated to the new names.

## Test Coverage

VisualGen unit suites under `tests/unittest/_torch/visual_gen/` (args, loader, pipeline, transformer, attention, per-model) and integration tests (`tests/integration/defs/examples/test_visual_gen.py`, `tests/integration/defs/perf/test_visual_gen_perf_sanity.py`) are updated to the new schema. Verified on GB200.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

## Release Notes

* **Configuration Schema Updates**
  * Consolidated visual generation configuration structure with renamed sections (`attention_config`, `parallel_config`, `cache_config`, `cuda_graph_config`)
  * Standardized parallelism field names (`cfg_size`, `ulysses_size` replacing `dit_cfg_size`, `dit_ulysses_size`)

* **CLI Updates**
  * Renamed command-line argument from `--extra_visual_gen_options` to `--visual_gen_args` for model serving
  * Updated documentation examples to reflect new configuration schema

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/TensorRT-LLM/pull/14175?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2_two_stages.py [hunk 1]
```diff
@@ -580,7 +580,13 @@ def _restore_lora_state(
 # ---------------------------------------------------------------------------
 
 
-@register_pipeline("LTX2TwoStagesPipeline")
+@register_pipeline(
+    "LTX2TwoStagesPipeline",
+    # Variant of LTX2Pipeline resolved at load time — no canonical
+    # HuggingFace id of its own and no separate defaults; the parent
+    # LTX2Pipeline entry is the one users discover.
```
Comment (zhenhuaw-me) [inline tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2_two_stages.py]: The   "spatial_upsampler_path" and "distilled_lora_path" should move to here - they are 2stage only config. And we should have the same HF id as the LTX2Pipeline.

Drop the comment

File: examples/visual_gen/serve/configs/ltx2.yml [hunk 2]
```diff
@@ -1,8 +1,7 @@
-text_encoder_path: google/gemma-3-12b-it
-cache:
+cache_config:
   cache_backend: cache_dit
```
Comment (zhenhuaw-me) [inline examples/visual_gen/serve/configs/ltx2.yml:2]: @o-stoner If it's expected that we enable cache_dit for ltx2 but no further cache_dit config?

**Review by yufeiwu-nv — APPROVED** [review]

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py [hunk 3]
```diff
@@ -79,7 +79,16 @@
 )
 
 
-@register_pipeline("WanPipeline")
+@register_pipeline(
+    "WanPipeline",
+    hf_ids=[
```
Comment (zhenhuaw-me) [inline tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py:83]: @o-stoner Can you help to review the support HF model ID list? Thanks!

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_i2v.py [hunk 4]
```diff
@@ -79,7 +79,15 @@
 )
 
 
-@register_pipeline("WanImageToVideoPipeline")
+@register_pipeline(
+    "WanImageToVideoPipeline",
+    hf_ids=[
+        "Wan-AI/Wan2.1-I2V-14B-480P-Diffusers",
```
Comment (zhenhuaw-me) [inline tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_i2v.py:84]: @o-stoner Can you help to review the support HF model ID list? Thanks!

File: tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2_two_stages.py [hunk 5]
```diff
@@ -580,13 +580,31 @@ def _restore_lora_state(
 # ---------------------------------------------------------------------------
 
 
-@register_pipeline("LTX2TwoStagesPipeline")
+@register_pipeline(
+    "LTX2TwoStagesPipeline",
+    hf_ids=[
+        "Lightricks/LTX-2",
```
Comment (zhenhuaw-me) [inline tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2_two_stages.py]: @yibinl-nvidia Do we support other variants? What's ModelOpt's plan about FP4 ckpt release?

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py (hunk 3 above)
Comment (o-stoner) [reply tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py:83]: These should be correct for now - we will add to this list to support FastVideo VSA models from HF

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_i2v.py (hunk 4 above)
Comment (o-stoner) [reply tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_i2v.py:84]: LGTM

File: tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2_two_stages.py (hunk 5 above)
Comment (zhenhuaw-me) [reply tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2_two_stages.py]: I suppose not.

**Review by kaiyux — APPROVED** [review]
Approving on behalf of `trtllm-bench-reviewers`

**Review by karljang — APPROVED** [review]
I have confirmed the FLUX parts.

**Comment by luyiyun1021** [comment]
LTX part LGTM

File: tests/integration/defs/examples/test_visual_gen.py
Comment (QiJune) [inline tests/integration/defs/examples/test_visual_gen.py]: This LPIPS helper still uses the old VisualGenArgs schema: `PipelineConfig`, `checkpoint_path`, `device`, `dtype`, `pipeline`, `skip_warmup`, and old `torch_compile` fields. These now fail with the  strict public config.

File: tensorrt_llm/_torch/visual_gen/attention_backend/trtllm.py [hunk 6]
```diff
@@ -25,11 +25,11 @@
 
 from tensorrt_llm.mapping import Mapping
 from tensorrt_llm.models.modeling_utils import QuantConfig
+from tensorrt_llm.visual_gen.args import QuantAttentionConfig
```
Comment (QiJune) [inline tensorrt_llm/_torch/visual_gen/attention_backend/trtllm.py:28]: Nit: `QuantAttentionConfig` reads a bit awkward. Could we name this `AttentionQuantConfig` instead? It feels clearer that this is quantization config for attention.

**Review by QiJune — APPROVED** [review]
LGTM

File: tests/integration/defs/examples/test_visual_gen.py
Comment (zhenhuaw-me) [reply tests/integration/defs/examples/test_visual_gen.py]: Probably related to the rebase. Thanks for catching it. Fixed!

File: tensorrt_llm/_torch/visual_gen/attention_backend/trtllm.py (hunk 6 above)
Comment (zhenhuaw-me) [reply tensorrt_llm/_torch/visual_gen/attention_backend/trtllm.py:28]: It was following the style like `SkipSoftmaxAttentionConfig`. I will leave it as-is, please let me know if any concerns. Thanks!
FYI: we will add a `SparseAttentionConfig` sub config to `AttentionConfig` soon.

**Review by o-stoner — APPROVED** [review]
Wan LGTM

**Review by brb-nv — APPROVED** [review]
Approving on behalf of dynamo-devs.

File: tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2.py [hunk 7]
```diff
@@ -565,7 +565,32 @@ def _load_component_weights(
...
+        "text_encoder_path": "google/gemma-3-12b-it",
+        "spatial_upsampler_path": None,
+        "distilled_lora_path": None,
+    },
+    doc=(
+        "Lightricks LTX-Video. ``pipeline_config()`` returns the "
+        "superset of knobs. Setting both ``spatial_upsampler_path`` and "
+        "``distilled_lora_path`` selects the two-stage refinement "
+        "variant via ``resolve_variant()`` at load time; leaving either "
+        "as ``None`` stays on the one-stage variant."
```
Comment (yibinl-nvidia) [inline tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2.py]: This wording looks confusing. LTX-2 does not simply stay on one stage pipeline whenever either auxiliary path is `None`: the resolver will auto-discover the spatial upsampler and distilled LoRA from the checkpoint path, and will auto-upgrade to two stage pipeline when both checkpoints are provided or discovered.

Could we either update this doc to describe the actual resolution order or remove this part?

**Review by yibinl-nvidia — APPROVED** [review]
LGTM on the LTX-2, left a comment on the doc.

File: tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2.py (hunk 7 above)
Comment (zhenhuaw-me) [reply tensorrt_llm/_torch/visual_gen/models/ltx2/pipeline_ltx2.py]: Updated. Thanks for catching it!

**Comment by zhenhuaw-me** [comment]
/bot reuse-pipeline --comment "last CI passed and the new commit is doc update and passing local UTs"

File: tensorrt_llm/visual_gen/args.py [hunk 8]
```diff
@@ -12,15 +12,551 @@
...
+
+    Bare ``QuantAttentionConfig()`` is a valid finest-granularity recipe
+    (``qk_dtype="int8"``, ``v_dtype="e4m3"``, ``q=k=v=1``); tune
+    ``k_block_size`` up (4 or 16) for speed.
+
+    Unsupported recipes are rejected by ``AttentionConfig``'s validator
+    with a ``ValueError``.
+    """
+
+    qk_dtype: Literal["int8", "e4m3"] = Field(
```
Comment (chang-l) [inline tensorrt_llm/visual_gen/args.py]: can we use fp8 instead of e4m3 as the dtype?

File: examples/visual_gen/configs/wan2.2-t2v-fp4-1gpu.yaml [hunk 9]
```diff
@@ -18,10 +18,10 @@
 quant_config:
   quant_algo: NVFP4
   dynamic: true
-attention:
+attention_config:
   backend: VANILLA
```
Comment (chang-l) [inline examples/visual_gen/configs/wan2.2-t2v-fp4-1gpu.yaml:22]: How do we enable SageAttention now? Could we align this with the quant_config used by our linear layers above? For example, for mxfp8 and sage:

```
attention_config:
  backend: TRTLLM / CUTEDSL
  quant_config:
        dtype: mxfp8/fp8 (or quant_algo: mxfp8 / sage)

# or more fine-grained (like sage):
   quant_config:
          qk_dtype: int8
           v_dtype: fp8
```

File: tensorrt_llm/_torch/visual_gen/pipeline.py [hunk 10]
```diff
@@ -8,14 +8,14 @@
 import torch.nn as nn
 from pydantic import Field
 
+from tensorrt_llm._torch.visual_gen.pipeline_loader import PipelineComponent
```
Comment (chang-l) [inline tensorrt_llm/_torch/visual_gen/pipeline.py]: This import looks like it creates a circular import. `pipeline.py` now imports `PipelineComponent` from `pipeline_loader.py`, but `pipeline_loader.py` imports `.models`, and `models/__init__.py` imports `BasePipeline` back from `pipeline.py` before `BasePipeline` has been defined. Direct imports such as `from tensorrt_llm._torch.visual_gen.pipeline import BasePipeline` can fail with a partially initialized module. Please move `PipelineComponent` to a small leaf module, or back to a config/constants module, so core pipeline code does not depend on the loader/factory stack.

File: tensorrt_llm/commands/serve.py [hunk 11]
```diff
@@ -1043,21 +1047,16 @@ def _serve_llm():
                           served_model_name=served_model_name)
 
     def _serve_visual_gen():
-        extra_args = {}
-        if extra_visual_gen_options is not None:
-            with open(extra_visual_gen_options, 'r') as f:
-                extra_args = yaml.safe_load(f) or {}
-
-        diffusion_args = VisualGenArgs(**extra_args) if extra_args else None
+        parsed_visual_gen_args = (VisualGenArgs.from_yaml(visual_gen_args)
```
Comment (chang-l) [inline tensorrt_llm/commands/serve.py:1050]: This now feeds the YAML directly into strict `VisualGenArgs.from_yaml()`, while the CLI still accepts the legacy `--extra_visual_gen_options` alias. Existing YAML files for that old flag can use fields like `parallel`, `cache`, `attention`, `torch_compile`, `cuda_graph`, `checkpoint_path`, or top-level LTX fields, and those will now fail with Pydantic extra-field errors instead of migrating. Either remove/deprecate the legacy alias and docs, or normalize the old YAML shape before validation.

File: docs/source/models/visual-generation.md [hunk 12]
```diff
@@ -102,7 +102,7 @@ Programmatic usage via `VisualGenArgs.quant_config`:
 from tensorrt_llm import VisualGenArgs
 
 args = VisualGenArgs(
-    checkpoint_path="/path/to/model",
+    model="/path/to/model",
```
Comment (chang-l) [inline docs/source/models/visual-generation.md:105]: This file still documents some of the old VisualGen YAML shape. Updating this example to `model` is good, but the TeaCache section still shows `teacache.enable_teacache: true`; under the new `VisualGenArgs` schema that should be configured as something like `cache_config: {cache_backend: teacache, ...}`. Following the current doc through `trtllm-serve --visual_gen_args` will fail strict validation.

File: examples/visual_gen/configs/wan2.2-t2v-fp4-1gpu.yaml (hunk 9 above)
Comment (zhenhuaw-me) [reply examples/visual_gen/configs/wan2.2-t2v-fp4-1gpu.yaml:22]: It's the fine-grained case.

File: tensorrt_llm/commands/serve.py (hunk 11 above)
Comment (zhenhuaw-me) [reply tensorrt_llm/commands/serve.py:1050]: The CLI alias only renames the flag — both `--visual_gen_args` and `--extra_visual_gen_options` route to the same `visual_gen_args` variable that is parsed by `VisualGenArgs.from_yaml()`. The YAML-shape break you're describing is from the schema refactor in commit `5ad402fef3` and is independent of the alias; removing the alias wouldn't make old YAMLs work, and keeping it doesn't preserve the old shape.

Since `visual_gen_args` is prototype-status, the alias is just there as a one-release courtesy for the flag-name rename. Happy to drop it if you'd prefer.

File: docs/source/models/visual-generation.md (hunk 12 above)
Comment (zhenhuaw-me) [reply docs/source/models/visual-generation.md:105]: Fixed. Thanks for catching it!

File: tensorrt_llm/_torch/visual_gen/pipeline.py (hunk 10 above)
Comment (zhenhuaw-me) [reply tensorrt_llm/_torch/visual_gen/pipeline.py]: Moved to registery file to avoid.

File: tensorrt_llm/visual_gen/args.py (hunk 8 above)
Comment (zhenhuaw-me) [reply tensorrt_llm/visual_gen/args.py]: Makes sense. Switched to `fp8` and doc that it's `e4m3` in practice.

File: examples/visual_gen/configs/wan2.2-t2v-fp4-1gpu.yaml (hunk 9 above)
Comment (chang-l) [reply examples/visual_gen/configs/wan2.2-t2v-fp4-1gpu.yaml:22]: Got it. I meant `quant_config` in the sketch, but looking at `args.py`, the current PR schema uses `quant_attention_config` under `AttentionConfig`, i.e.,
```
 attention_config:
    backend: TRTLLM
    quant_attention_config:
```

Non-blocking: can we rename/alias that field to `quant_config` for consistency with the linear-layer config? (feel free to address it in a followup pr)
